### PR TITLE
fix(docker source): add retry delay

### LINKED ISSUE #3709: Add delay when retrying `docker` source failures
As noted in our Gitter chat. It appears the `docker` source does not back off on retries and emits these log messages continuously:

```
Sep 04 09:55:31.245  INFO vector::sources::docker: Stoped listening logs on docker container id=1be0db0d93a623922fff433c13c7ce082df9e01a08d978dc9e00559dd685d32d
Sep 04 09:55:31.245  INFO vector::sources::docker: Started listening logs on docker container id=1be0db0d93a623922fff433c13c7ce082df9e01a08d978dc9e00559dd685d32d
Sep 04 09:55:31.246 ERROR vector::sources::docker: docker API container logging error error=API responded with a 404 not found: {"message":"No such container: 1be0db0d93a623922fff433c13c7ce082df9e01a08d978dc9e00559dd685d32d"}
```

We should ensure the failures are properly backed off, and also think about the error messages we are emitting.

### PR DESCRIPTION
**Pull request by juchiast**
fixes #3709 

Signed-off-by: Duy Do <juchiast@gmail.com>

### CODE REVIEW

**Review by fanatid — APPROVED** [review]

**Comment by juchiast** [comment]
Added a `retry_backoff_secs` configuration, default is 2 secs.

File: src/sources/docker.rs [hunk 1]
```diff
@@ -396,7 +402,8 @@ impl DockerSource {
                                         .remove(&id)
                                         .expect("Every started ContainerId has it's ContainerState");
                                     if state.is_running() {
-                                        self.containers.insert(id.clone(), self.esb.start(id, Some(Duration::from_secs(5))));
+                                        let backoff = Some(self.backoff_duration.clone());
```
Comment (fanatid) [inline src/sources/docker.rs:405]: `.clone()` should not be required, std::time::Duration impl Copy

Comment (juchiast) [reply src/sources/docker.rs:405]: @fanatid Thanks! I didn't notice that `make check-clippy` fails. Is anyone on the action to move `EmbarkStudios/cargo-deny-action` somewhere else? It always fails at the moment and is distracting other checks.

Comment (fanatid) [reply src/sources/docker.rs]: It's strange that clippy did not catch this. I also receive a failed test for cargo deny (and master too), I'll try to fix it after 1-2 hours.

Comment (juchiast) [reply src/sources/docker.rs]: @fanatid clippy did catch this. I saw the check fails and thought it's cargo-deny-action and ignored it.

**Comment by ktff** [comment]
This looks nice, just the documentation needs to be added [here](https://github.com/timberio/vector/blob/master/.meta/sources/docker.toml.erb)

I'm also curious if `config/vector.spec.toml` is something that we should update by hand or is it auto generated?

**Comment by juchiast** [comment]
> I'm also curious if config/vector.spec.toml is something that we should update by hand or is it auto generated?

Look like it's updated by hand, most configurations in the code do not have comments so they cannot be used to generate the spec file.

**Comment by jszwedko** [comment]
@juchiast @ktff I believe `vector.spec.toml` is generated by https://github.com/timberio/vector-generate from [`vector.spec.toml.erb`](https://github.com/timberio/vector/blob/master/config/vector.spec.toml.erb).

**Comment by binarylogic** [comment]
Yes, only the files in the root `.meta` should need adjusting for documentation, nothing else. So we should update the [`docker.toml.erb`](https://github.com/timberio/vector/blob/master/.meta/sources/docker.toml.erb) for this.

**Pull request merged**